common/src/leap/soledad/common/couch.py

```diff
@@ -1474,14 +1506,32 @@ def ensure_database(self, dbname):
         databases. But, in case it ever does, by raising an exception here we
         have one more guarantee that no modified client will be able to
         enforce creation of a database when syncing.
+        This can only be done by using a shell script with proper validation
+        and permissions (create_cmd).
 
         :param dbname: The name of the database to ensure.
         :type dbname: str
 
         :raise Unauthorized: Always, because Soledad server is not allowed to
                              create databases.
+
+        :return: The CouchDatabase object and its replica_uid.
+        :rtype: (CouchDatabase, str)
         """
-        raise Unauthorized()
+        if not self.create_cmd:
+            raise Unauthorized()
+        else:
+            code, out = exec_validated_cmd(self.create_cmd, dbname,
+                                           validator=is_db_name_valid)
+            if code is not 0:
+                logger.error("""
+                    Error while creating database (%s) with (%s) command.
+                    Output: %s
+                    Exit code: %d
+                    """ % (dbname, self.create_cmd, out, code))
+                raise Unauthorized()
```

Comment:
What will be the outcome of a temporary error error with couchdb, this call will fail. Will the client sync fail and then will it retry to create the database in the next sync attempt?